quarkScriptAgent.py

```diff
@@ -0,0 +1,144 @@
+import os
+import re
+
+from termcolor import colored
+
+from langchain_openai import ChatOpenAI
+from langchain.agents import tool, AgentExecutor
+from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
+from langchain.agents.output_parsers.openai_tools import OpenAIToolsAgentOutputParser
+from langchain.agents.format_scratchpad.openai_tools import (
+    format_to_openai_tool_messages,
+)
+
+from quark.script import Rule, _getQuark, QuarkResult
+
+if "OPENAI_API_KEY" not in os.environ:
+    api_key = input("OpenAI API Key: ")
+    os.environ["OPENAI_API_KEY"] = api_key
+
+
+@tool
+def defineRuleByJson():
+    """
+    This instance defines the detection rule
+    by loading the rule from a JSON.
+    """
+
+    global ruleInstance
+    ruleInstance = Rule("rule.json")
+
+    return "Rule defined successfully"
+
+
+@tool
+def runQuarkAnalysis(samplePath):
+    """
+    Given detection rule and target sample,
+    this instance runs the basic Quark.
```

Comment:
I suggest replacing `the basic Quark` with `the Quark analysis` to make the docstring consistent with the tool's name.